Library/Homebrew/utils/diskusage.rb

```diff
@@ -0,0 +1,57 @@
+require "find"
+
+module Utils
+  class DiskUsage
+    attr_reader :size_in_bytes, :file_count
+
+    def initialize(path, excludes = [])
+      @size_in_bytes = 0
+      @file_count = 0
+      @path = path
+      @excludes = [".DS_Store"] + excludes
+      begin
+        Find.find(@path).each do |f|
+          if !File.directory?(f) and !f.match(/#{@excludes.join("|")}/)
+            @file_count += 1
+            @size_in_bytes += File.size(f)
+          end
+        end
+      rescue Errno::ENOENT
+        raise "DiskUsage: No such path `#{@path}'"
+      end
+    end
+
+    def to_s
+      out = "#{@path} ("
+      out << "#{number_readable(@file_count)} files, " if @file_count > 1
+      out << "#{disk_space_readable(@size_in_bytes)})"
+    end
+
+    def self.disk_space_readable(size_in_bytes)
+      len = size_in_bytes.to_s.length
+      case
+      when len > 9
+        sym, unit = ["G", 1_073_741_824]
+      when len > 6
+        sym, unit = ["M", 1_048_576]
+      when len > 3
+        sym, unit = ["K", 1_024]
+      else
+        sym, unit = ["B", 1]
+      end
+      "%.1f#{sym}" % [size_in_bytes.to_f / unit]
+    end
+
+    def disk_space_readable(size_in_bytes)
+      self.class.disk_space_readable(size_in_bytes)
+    end
+
+    def self.number_readable(number)
+      number.to_s.reverse.split("").each_slice(3).to_a.map {|x| x.reverse.join}.reverse.join(",")
```

Comment:
Quite some heavy lifting for what should be a simple string operation. I'd suggest:

```ruby
str = number.to_i.to_s
(str.size - 3).step(1, -3) { |i| str.insert(i, ",") }
str
```